### PR TITLE
update dependency and golang

### PR DESCRIPTION
**Pull request by jwsui**
Handle dependabot alerts. Update `k8s.io/kubernetes` and its related modules
including `cluster-network-operator`/`controller-runtime`/`client-go`. Also,
replace `github.com/dgrijalva/jwt-go`. Also, update golang from 1.13 to 1.16.

### CODE REVIEW

File: controllers/antreainstall_controller.go [hunk 1]
```diff
@@ -357,7 +357,7 @@ func (r *AntreaInstallReconciler) SetupWithManager(mgr ctrl.Manager) error {
 // +kubebuilder:rbac:urls=/agentinfo;/addressgroups;/appliedtogroups;/networkpolicies;/ovsflows;/ovstracing;/podinterfaces,verbs=get
 // +kubebuilder:rbac:groups=security.openshift.io,resources=securitycontextconstraints,resourceNames=hostnetwork,verbs=use
 
-func (r *AntreaInstallReconciler) Reconcile(request ctrl.Request) (reconcile.Result, error) {
+func (r *AntreaInstallReconciler) Reconcile(cxt context.Context, request ctrl.Request) (reconcile.Result, error) {
```
Comment (jwsui) [inline controllers/antreainstall_controller.go:360]: Updated due to interface change.

File: go.mod [hunk 2]
```diff
@@ -1,20 +1,25 @@
...
+	github.com/go-logr/logr v0.4.0
+	github.com/onsi/gomega v1.13.0
+	github.com/openshift/api v0.0.0-20211203184245-fb6f933bb8d5
+	github.com/openshift/cluster-network-operator v0.0.0-20220125073742-83efd7f8d510
+	github.com/openshift/library-go v0.0.0-20210708173104-7e7d216ed91c
+	gopkg.in/yaml.v2 v2.4.0
+	k8s.io/api v0.22.1
+	k8s.io/apimachinery v0.22.1
+	k8s.io/client-go v0.22.1
+	sigs.k8s.io/controller-runtime v0.9.0-beta.0
```
Comment (jwsui) [inline go.mod:17]: Upgrade `k8s.io/kuberbetes` related modules, including `github.com/openshift/cluster-network-operator`/`k8s.io/client-go`/`sigs.k8s.io/controller-runtime`.

File: go.mod [hunk 3]
```diff
@@ -1,20 +1,25 @@
...
 )
 
-replace k8s.io/client-go => k8s.io/client-go v0.18.6
+replace (
+	github.com/dgrijalva/jwt-go => github.com/golang-jwt/jwt v3.2.1+incompatible
```
Comment (jwsui) [inline go.mod:21]: migrate to `golang-jwt`.
[https://pkg.go.dev/github.com/golang-jwt/jwt#section-readme](https://pkg.go.dev/github.com/golang-jwt/jwt#section-readme)

File: go.mod [hunk 4]
```diff
@@ -1,20 +1,25 @@
...
+	sigs.k8s.io/cluster-api-provider-aws => github.com/openshift/cluster-api-provider-aws v0.2.1-0.20201125052318-b85a18cbf338
```
Comment (jwsui) [inline go.mod:22]: These three replacements are need by `cluster-network-operator`.
[https://github.com/openshift/cluster-network-operator/blob/master/go.mod](https://github.com/openshift/cluster-network-operator/blob/master/go.mod)

**Comment by jwsui** [comment]
The check was failed because it used golang-1.13 and golang was updated to 1.16 in this patch.

**Review by dantingl — APPROVED** [review]

**Comment by jwsui** [comment]
Check passed after upgrade go-1.16 in workflow, merge pr.

**Pull request merged**